packages/pieces/community/slack/src/lib/triggers/new-channel.ts

```diff
@@ -1,81 +1,66 @@
-import {
-  createTrigger,
-  PiecePropValueSchema,
-  TriggerStrategy,
-} from '@activepieces/pieces-framework';
+import { WebClient } from '@slack/web-api';
 
-const polling: Polling<
-  PiecePropValueSchema<typeof slackAuth>,
-  Record<string, never>
-> = {
-  strategy: DedupeStrategy.TIMEBASED,
-  items: async ({ auth, propsValue, lastFetchEpochMS }) => {
-    let cursor;
-    const channels: any[] = [];
-    do {
-      const request: HttpRequest = {
-        method: HttpMethod.GET,
-        url: 'https://slack.com/api/conversations.list',
-        queryParams: {
-          types: 'public_channel,private_channel',
-          limit: '200',
-          cursor: cursor ?? '',
-        },
-
-        authentication: {
-          type: AuthenticationType.BEARER_TOKEN,
-          token: auth.access_token,
-        },
-      };
-
-      const response = await httpClient.sendRequest<{
-        channels: any[];
-        response_metadata: { next_cursor: string };
-      }>(request);
-
-      channels.push(...response.body.channels);
-      cursor = response.body.response_metadata.next_cursor;
-    } while (cursor !== '' && channels.length < 600);
-
-    return channels.map((channel: any) => ({
-      epochMilliSeconds: channel.created * 1000,
-      data: channel,
-    }));
+const sampleData = {
+  type: 'channel_created',
+  channel: {
+    id: 'C024BE91L',
+    name: 'fun',
+    created: 1360782804,
+    creator: 'U024BE7LH',
   },
 };
 
-export const newChannelTrigger = createTrigger({
+export const channelCreated = createTrigger({
   auth: slackAuth,
-  name: 'slack-new-channel',
-  displayName: 'New Channel',
-  description: 'Triggers when a new channel is created.',
-  type: TriggerStrategy.POLLING,
+  name: 'channel_created',
+  displayName: 'Channel created',
+  description: 'Triggers when a channel is created',
   props: {},
-  async onEnable(context) {
-    const { store, auth, propsValue } = context;
-    return await pollingHelper.onEnable(polling, { store, auth, propsValue });
+  type: TriggerStrategy.APP_WEBHOOK,
+  sampleData: sampleData,
+  onEnable: async (context) => {
+    // Older OAuth2 has team_id, newer has team.id
+    const teamId =
+      context.auth.data['team_id'] ?? context.auth.data['team']['id'];
+    context.app.createListeners({
+      events: ['channel_created'],
+      identifierValue: teamId,
+    });
   },
-  async onDisable(context) {
-    const { store, auth, propsValue } = context;
-    return await pollingHelper.onDisable(polling, { store, auth, propsValue });
+  onDisable: async (context) => {
+    // Ignored
   },
-  async test(context) {
-    const { store, auth, propsValue } = context;
-    return await pollingHelper.test(polling, { store, auth, propsValue });
+  test: async (context) => {
+    const client = new WebClient(context.auth.access_token);
+    const response = await client.conversations.list({
+      exclude_archived: true,
+      limit: 10,
+      types: 'public_channel',
```

Comment:
@AdamSelene should we include other types?

Reply:
no strong opinion, it's only for testing anyway